crates/matrix-sdk/src/room/reply.rs

```diff
@@ -111,6 +116,45 @@ impl Room {
         )
         .await
     }
+
+    /// Create a new reply event for the specified [`RepliedToInfo`] and
+    /// content.
+    ///
+    /// The event can then be sent with [`Room::send`] or a
+    /// [`crate::send_queue::RoomSendQueue`].
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content to reply with
+    /// * `replied_to_info` - Information about the event to reply to
+    /// * `enforce_thread` - Whether to enforce a thread relation
+    #[instrument(skip(self, content), fields(room = %self.room_id()))]
+    pub fn make_reply_event_with_info(
+        &self,
+        content: RoomMessageEventContentWithoutRelation,
+        replied_to_info: RepliedToInfo,
+        enforce_thread: EnforceThread,
+    ) -> RoomMessageEventContent {
+        make_reply_event_with_info(
+            self.room_id(),
+            self.own_user_id(),
+            content,
+            replied_to_info,
+            enforce_thread,
+        )
+    }
+
+    /// Retrieve the information needed to reply to an event.
+    ///
+    /// # Arguments
+    ///
+    /// * `event_id` - ID of the event to reply to
+    pub async fn replied_to_info_from_event_id(
+        &self,
+        event_id: &EventId,
+    ) -> Result<RepliedToInfo, ReplyError> {
+        replied_to_info_from_event_id(self, event_id).await
+    }
```

Comment:
I extracted these methods so that, when the requested event cannot be replied to, we can fail early and before starting to process any media. It does, unfortunately, make the interface a little more complicated.

Reply:
Yeah, this gets us back to a place where responding is a two-process step:

1. get the replied-to info
2. call the `make_reply_event_with_info` method

We could avoid this new method and the one above, by reusing `make_reply_event`. I think this would make `Room::make_attachment_event` async+fallible (while it's not any of these two right now), but that seems fine; both callers are already async+fallible too.

Can you try this out please? If it turns out to be too complicated, we can consider a two-steps process like now, but having a super streamlined API would be sweet.

Reply:
Yeah, this seems to work well actually. 👍